### ISSUE TITLE
ref-readonly returns in local functions are giving an error

### ISSUE THREAD

**Issue by OmarTawfik** [dotnet/roslyn#18357]
```csharp
class Test
{
	public void Outer()
	{
		ref readonly int Inner(ref readonly int x)
		{
			return ref x;
		}
	}
}
```

**Expected**: No errors.
**Actual**: error CS8406: Cannot use variable 'in int' as a ref or out value because it is a readonly variable
